pkg/deviceplugin/plugin.go

```diff
@@ -0,0 +1,133 @@
+package deviceplugin
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+
+	"github.com/kubevirt/macvtap-cni/pkg/util"
+)
+
+const (
+	tapPath         = "/dev/tap"
+	suffix          = "Mvp"
+	defaultCapacity = 100
+)
+
+type MacvtapDevicePlugin struct {
+	Name        string
+	Master      string
+	Mode        string
+	Capacity    int
+	stopWatcher chan struct{}
+}
+
+func NewMacvtapDevicePlugin(name string, master string, mode string, capacity int) *MacvtapDevicePlugin {
+	return &MacvtapDevicePlugin{
+		Name:        name,
+		Master:      master,
+		Mode:        mode,
+		Capacity:    capacity,
+		stopWatcher: make(chan struct{}),
+	}
+}
+
+func (mdp *MacvtapDevicePlugin) generateMacvtapDevices() []*pluginapi.Device {
+	var macvtapDevs []*pluginapi.Device
+
+	var capacity = mdp.Capacity
+	if capacity <= 0 {
+		capacity = defaultCapacity
+	}
+
+	for i := 0; i < capacity; i++ {
+		name := fmt.Sprint(mdp.Name, suffix, i)
```

Comment:
won't the suffix create invalid device names - e.g. `/dev/tapMvpX` ?

Reply:
This is the name of the interface, not the tap char device. The tap char device path is passed to  kubelet in the `Allocate` request.

Reply:
Right; got it.

Reply:
Ah.

Sure.